libs/fetcher.client.js

```diff
@@ -326,6 +326,14 @@ Fetcher.prototype = {
             .params(params)
             .clientConfig(clientConfig)
             .end(callback)
+    },
+
+    /**
+     * Update options
+     * @method updateOptions
+     */
+    updateOptions: function (options) {
+        this.options = lodash.merge(this.options, options);
```

Comment:
we will still need `lodash.merge` here since we might need to deeply merge context